modules/subscriptionService/src/main/scala/service/SubscriptionService.scala

```diff
@@ -0,0 +1,55 @@
+package service
+
+import cats.effect.Sync
+import userDbAlgebra.UserAlgebra
+import subscriptionAlgebra.SubscriptionServiceAlgebra
+import utils.Types.{GetSubscriptionData, Id, PostSubscriptions}
+import doobie.ConnectionIO
+import doobie.implicits._
+import doobie.util.transactor.Transactor
+
+sealed trait UserErrors                  extends Exception
+case class UserNotFound(message: String) extends UserErrors
+
+object SubscriptionService {
+  def implementation[F[_]: Sync](
+      userAlgebra: UserAlgebra[ConnectionIO],
+      subscriptionAlgebra: SubscriptionServiceAlgebra[ConnectionIO],
+      transactor: Transactor[F]
+  ): SubscriptionServiceAlgebra[F] = new SubscriptionServiceAlgebra[F] {
+    def get(id: Id): F[Option[GetSubscriptionData]] = {
+      for {
+        user <- userAlgebra.findUser(id)
+        optionOfGetSubscription <- user match {
+          case None =>
+            Sync[ConnectionIO].raiseError(
+              UserNotFound("User does not Exit")
+            )
+          case Some(user) => subscriptionAlgebra.get(user.id)
+        }
+      } yield optionOfGetSubscription: Option[GetSubscriptionData]
+    }.transact(transactor)
+
+    def post(id: Id, subscriptions: PostSubscriptions): F[Unit] = (for {
+      checkIfTheUserExits <- userAlgebra.findUser(id)
+      _ <- checkIfTheUserExits match {
```

Comment:
Did you solve the problem with the inferred `Any`?

```scala
  "-Xlint:infer-any",                  // Warn when a type argument is inferred to be `Any`.
```
I am not sure if you are using it already, but I highly recommend using this plugin: 

```scala
addSbtPlugin("io.github.davidgregory084" % "sbt-tpolecat"        % "0.1.20")
```

Reply:
@moleike  I resolved it by specifying the return type on the `yield`.
Also I will have a loot at the plugin thank you 😊

Reply:
@moleike I am not having a lot of success with this plugin 😂